src/rust/inetstack/protocols/layer4/tcp/established/sender.rs

```diff
@@ -459,26 +458,36 @@ impl Sender {
             // Check and update send window if necessary.
             self.update_send_window(header);
 
-            if header.ack_num == send_next {
-                // This segment acknowledges everything we've sent so far (i.e. nothing is currently outstanding).
-                // Since we no longer have anything outstanding, we can turn off the retransmit timer.
-                debug_assert_eq!(self.unacked_queue.is_empty(), true);
-                self.retransmit_deadline_time_secs.set(None);
-            } else {
-                // Update the retransmit timer.  Some of our outstanding data is now acknowledged, but not all.
-                // TODO: This looks wrong.  We should reset the retransmit timer to match the deadline for the
-                // oldest still-outstanding data.  The below is overly generous (minor efficiency issue).
-                let deadline: Instant = now + self.rto_calculator.rto();
-                self.retransmit_deadline_time_secs.set(Some(deadline));
+            // Reset the retransmit timer if necessary. If there is more data that hasn't been acked, then set to the
+            // next segment deadline, otherwise, do not set.
+            let retransmit_deadline_time_secs: Option<Instant> = self.update_retransmit_deadline(now);
+            #[cfg(debug_assertions)]
```

Comment:
The macro already tests for this, so this can be removed. Unless we want to conditionally compile out any trace of this macro in the codebase when built in the `release` mode. If so, then we should do the same at all references of this macro.

Here's the macro def.:

<img width="370" alt="image" src="https://github.com/user-attachments/assets/7d08d84d-c4f9-4c3b-95d5-ef9ce3d4c966">

Reply:
Ah ok!

Reply:
We discussed offline, this will be fixed in consequent PR.